Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,790 @@
+The precise handling of errors will be specific to the implementing program but should use whatever error mechanisms exist already, for example Yarn Spinner throws normal C# exceptions for its errors.
+
+Dates, spelling, and numbers are to be in Australian English.
+
+### Modifying This Specification
+
+Once this specification is complete it is set and unchanging.
+Modifications beyond clarifications won't be allowed.
+To make changes to the language specification a new version of the language will need to be created.
+
+This is to ensure implementing programs can conform to a language version and not have to worry about it changing after the fact.
+
+## File Format
+
+Yarn files must be a UTF-8 text file without BOM set.
+File extension should be `.yarn`.
+
+### Lines
+
+The *line* is the common unit that comprises all elements in a yarn file.
+A line is a series of characters terminated by the *new line* symbol.
+The new line symbol should be the `\n` character.
+The new line must be the same throughout the project regardless of the chosen new line symbol.
+
+The following Yarn file contains four lines (in order); one header tag line, one header delimiter line, one body dialogue line, one body delimiter line.
+```yarn
+Title:Start
+---
+This is some text
+===
+```
+
+### Whitespace
+
+*Whitespace* is any non-visible character with a width greater than 0.
+Common whitespace encountered include the space and the tab.
+
+Whitespace for lines of dialogue plays no role but has significant syntactic important for Options.
+
+### Project
+
+The yarn *project* is all yarn files that are intended to be associated with one another.
+While there is nothing stopping a writer from placing all nodes into one big file (or even making one giant node) it is common to break them up over multiple files.
+The project is all of these files collected and processed by the implementing program together.
+
+### Comments
+
+A comment is a line that starts with the `//` symbol.
+All text from the start of the comment to the end of the line must be ignored.
+A comment starting in the middle of another line ends that line at the point the `//` symbol is encountered.
+That line is assumed to have finished at that point as if the comment was not there.
+Comments must not impact the rest of the lines or have any impact on the resulting yarn program.
+ 
+### Identifiers
+
+Throughout various points in this document identifiers are mentioned, the rules for these are shared across all stages of the Yarn project.
+
+An *identifier* is any of the following symbols: an upper or lowercase letter A to Z, an underscore (`_`), a noncombining alphanumeric Unicode character in the Basic Multilingual Plane, or a character outside the Basic Multilingual Plane that isn't in the Private Use Area.
+After the first character digits, a period (`.`), and combining Unicode characters are also allowed.
+But not another `$` symbol.
+The minimum and maximum length of identifiers is unspecified.
```

Comment:
Minimum is 1 (not including the $ that starts a variable identifier)

Reply:
That's already encoded in the spec by making it list specific allowed characters instead of "everything but x,y,z".
Whitespace and other non-visible characters don't match the allowed characters, neither does nothing.